### PR TITLE
Hw first step in security

### PR DESCRIPTION
**Pull request by AdamMudrak**

### CODE REVIEW

File: src/main/java/com/example/onlinebookshop/dto/book/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.Min;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BookDto {
+    @Schema(name = Constants.ID, example = Constants.ID_EXAMPLE)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/dto/book/BookDto.java]: Is this a response? If so, do we need all these annotation here?

File: src/main/java/com/example/onlinebookshop/dto/user/UserResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.example.onlinebookshop.constants.Constants;
+import com.example.onlinebookshop.constants.UserDtoConstants;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UserResponseDto {
+    @Schema(name = Constants.ID, example = Constants.ID_EXAMPLE)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/dto/user/UserResponseDto.java]: The same question

File: src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java]: It's better to use 
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            throw new RegistrationException("User with email "
+                    + requestDto.getEmail() + " already exists");
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/services/impl/UserServiceImpl.java]: Can mapper do this logic?

File: src/main/java/com/example/onlinebookshop/controller/BookController.java [hunk 5]
```diff
@@ -30,23 +32,24 @@
 @Validated
 @RequiredArgsConstructor
 @RestController
-@Tag(name = Constants.BOOK_API_NAME,
-        description = Constants.BOOK_API_DESCRIPTION)
+@Tag(name = BookConstants.BOOK_API_NAME,
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/controller/BookController.java]: If you use constant, please use import static 
```suggestion
@Tag(name = BOOK_API_NAME,
```

Check all places

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's improve your solution!

File: src/main/java/com/example/onlinebookshop/dto/book/BookSearchParametersDto.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package com.example.onlinebookshop.dto.book;
+
+import static com.example.onlinebookshop.constants.BookDtoConstants.AUTHOR_DESCRIPTION;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookshop/dto/book/BookSearchParametersDto.java:3]: After you refactored code according my comment, I think maybe it's better to use
import static com.example.onlinebookshop.constants.BookDtoConstants.*; ? Or you can't pass checkstyle in this way?

**Review by Elena-Bruyako — APPROVED** [review]
Approve, but need to think how make imports more elegant - see the comment

File: src/main/java/com/example/onlinebookshop/dto/book/BookSearchParametersDto.java (hunk 6 above)
Comment (AdamMudrak) [reply src/main/java/com/example/onlinebookshop/dto/book/BookSearchParametersDto.java:3]: Unfortunately, yes, *-style import is not allowed by mate checkstyle. It this case, what do we choose? Constants.SOME_CONSTANT style or just many static imports?

**Pull request merged**